report_compassion/models/partner_communication.py

```diff
@@ -40,11 +41,69 @@ def send(self):
         Update the count of succes story prints when sending a receipt.
         :return: True
         """
-        print_bvr = self.filtered(lambda j: j.send_mode == 'physical' and
-                                  j.product_id)
-        print_bvr.write({'report_id': self.env.ref(
-            'report_compassion.report_a4_bvr').id})
-        return super().send()
+        bvr_to_send = self.filtered(lambda j: j.send_mode == 'digital'
+                                    and j.product_id)
+        bvr_to_print = self.filtered(lambda j: j.send_mode == 'physical'
+                                     and j.product_id)
+        bvr_both = self.filtered(lambda j: j.send_mode == 'both'
+                                 and j.product_id)
+
+        if bvr_both:
+            origin = self.env.context.get('origin')
+            for bvr in bvr_both:
+                if origin == 'both_email' or \
+                        (origin == 'both_print' and
+                         bvr.report_id !=
+                         self.env.ref('report_compassion.report_a4_bvr')):
+                    # email part
+                    self._put_bvr_in_attachments(
+                        bvr, background=origin == 'both_email')
+
+            return super(PartnerCommunication, bvr_both).send()
+
+        if bvr_to_send:
+            for bvr in bvr_to_send:
+                self._put_bvr_in_attachments(bvr, background=True)
+            return super(PartnerCommunication, bvr_to_send).send()
+
+        if bvr_to_print:
+            for bvr in bvr_to_print:
+                if bvr.report_id != self.env.ref('report_compassion.report_a4_bvr'):
+                    self._put_bvr_in_attachments(bvr, background=False)
+            return super(PartnerCommunication, bvr_to_print).send()
+
+    def _put_bvr_in_attachments(self, bvr, background):
+        pdf_download = self._generate_pdf_data(bvr, background)
+        bvr.attachment_ids.unlink()
```

Comment:
Why are you removing other attachments ? I don't think it's necessary.

Reply:
I did it because I had a bug otherwise but I corrected and refactored that part so it's ok now